CNF-6351(TELCODOCS-1003) OVN-Kubernetes attached to SR-IOV single NIC dual port (#55768)
openshift/openshift-docs: modules/ipi-install-configuring-host-dual-network-interfaces-in-the-install-config.yaml-file.adoc

.Prequisites

* Configure a `PTR` DNS record with a valid hostname for each node with a static IP address.
* Install the NMState CLI (`nmstate`).

[NOTE]
====
Errors in the YAML syntax might result in a failure to apply the network configuration. Additionally, maintaining the validated YAML syntax is useful when applying changes using Kubernetes NMState after deployment or when expanding the cluster.
====

.Procedure

. Add the NMState configuration to the `networkConfig` field to hosts within the `install-config.yaml` file:
+
[source,yaml]
----
      hosts:
      - hostname: worker-1
         interfaces:
          - name: eno1
            macAddress: 0c:42:a1:55:f3:06
          - name: eno2
            macAddress: 0c:42:a1:55:f3:07
         networkConfig: <1>
           interfaces: <2>
           - name: eno1     <3>
             type: ethernet <4>
             state: up
             mac-address: 0c:42:a1:55:f3:06
             ipv4:
               enabled: true
               dhcp: false  <5>
             ethernet:
               sr-iov:
                total-vfs: 2 <6>
             ipv6:

>                enabled: false

Reviewer: ```suggestion
               enabled: false
               dhcp: false  <5>

```

           - name: sriov:eno1:0
             type: ethernet
             state: up  <7>
             ipv4:
               enabled: false <8>
             ipv6:
               enabled: false
           - name: sriov:eno1:1
             type: ethernet
             state: down
           - name: eno2
             type: ethernet
             state: up
             mac-address: 0c:42:a1:55:f3:07
             ipv4:
               enabled: true
             ethernet:
               sr-iov:
               total-vfs: 2
             ipv6:
               enabled: false
           - name: sriov:eno2:0
             type: ethernet
             state: up
             ipv4:
               enabled: false
             ipv6:
               enabled: false
           - name: sriov:eno2:1
             type: ethernet
             state: down
           - name: bond0
             type: bond
             state: up
             min-tx-rate: 100 <9>
             max-tx-rate: 200 <10>
             link-aggregation:
               mode: active-backup <11>
               options:
                 primary: sriov:eno1:0 <12>

>                port:
>                 - sriov:eno1:0
>                 - sriov:eno2:0

Reviewer: ```suggestion
               port:
                 - sriov:eno1:0
                 - sriov:eno2:0
```

               ipv4:

>                  address:
>                  - ip: 10.19.16.57 <13>
>                    prefix-length: 23

Reviewer: ```suggestion
                 address:
                   - ip: 10.19.16.57 <13>
                     prefix-length: 23
```

                 dhcp: false
                 enabled: true
              ipv6:
                enabled: false
              dns-resolver:
                config:

>                   server:
>                   - 10.11.5.160
>                   - 10.2.70.215

Reviewer: ```suggestion
                  server:
                    - 10.11.5.160
                    - 10.2.70.215
```

//
// installing_bare_metal_ipi/ipi-install-installation-workflow.adoc

:_content-type: PROCEDURE
[id="configuring-host-dual-network-interfaces-in-the-install-config-yaml-file_{context}"]
= Optional: Configuring host network interfaces for dual port NIC

:FeatureName: Support for Day 1 operations associated with enabling NIC partitioning for SR-IOV devices
include::snippets/technology-preview.adoc[leveloffset=+1]

Before installation, you can set the `networkConfig` configuration setting in the `install-config.yaml` file to configure host network interfaces using NMState to support dual port NIC.

.Prequisites

* Configure a `PTR` DNS record with a valid hostname for each node with a static IP address.
* Install the NMState CLI (`nmstate`).

[NOTE]
====
Errors in the YAML syntax might result in a failure to apply the network configuration. Additionally, maintaining the validated YAML syntax is useful when applying changes using Kubernetes NMState after deployment or when expanding the cluster.
====

.Procedure

. Add the NMState configuration to the `networkConfig` field to hosts within the `install-config.yaml` file:
+
[source,yaml]
----
hosts:

> - hostname: worker-1
>    interfaces:

Reviewer: Invalid indentation.
```suggestion
- hostname: worker-1
  interfaces:
```

// This is included in the following assemblies:
//
// installing_bare_metal_ipi/ipi-install-installation-workflow.adoc

:_content-type: PROCEDURE
[id="configuring-host-dual-network-interfaces-in-the-install-config-yaml-file_{context}"]
= Optional: Configuring host network interfaces for dual NIC

Before installation, you can set the `networkConfig` configuration setting in the `install-config.yaml` file to configure host network interfaces using NMState to support dual NIC.

.Prequisites

* Configure a `PTR` DNS record with a valid hostname for each node with a static IP address.
* Install the NMState CLI (`nmstate`).

> 

Reviewer: ```suggestion
NOTE
Errors in the YAML syntax might result in a failure to apply the network configuration. Additionally, maintaining the validated YAML syntax is useful when applying changes using Kubernetes NMState after deployment or when expanding the cluster.
```

Suggestion to add this note (same as from "Optional: Configuring host network interfaces")

.Procedure

. Use the `networkConfig` configuration setting by adding the NMState configuration to hosts within the `install-config.yaml` file:
+
[source,yaml]
----
hosts:
 - hostname: worker-1
    interfaces:
     - name: eno1
       macAddress: 0c:42:a1:55:f3:06
     - name: eno2
       macAddress: 0c:42:a1:55:f3:07
    networkConfig: <1>

>       interfaces: <2>

Reviewer: Apply the same suggestions as mentioned in `modules/agent-install-sample-config-bond-sriov.adoc`

// This is included in the following assemblies:
//
// installing_bare_metal_ipi/ipi-install-installation-workflow.adoc

:_content-type: PROCEDURE
[id="configuring-host-dual-network-interfaces-in-the-install-config-yaml-file_{context}"]
= Optional: Configuring host network interfaces for dual port NIC

[IMPORTANT]
====

> Support for Day 1 operations associated with enabling NIC partitioning for SR-IOV devices is a Technology Preview feature. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.

Reviewer 2: Same comment about using TP snippet

For more information about the support scope of Red Hat Technology Preview
features, see https://access.redhat.com/support/offerings/techpreview/.
====

Before installation, you can set the `networkConfig` configuration setting in the `install-config.yaml` file to configure host network interfaces using NMState to support dual port NIC.

.Prequisites

* Configure a `PTR` DNS record with a valid hostname for each node with a static IP address.
* Install the NMState CLI (`nmstate`).

[NOTE]
====

> Errors in the YAML syntax might result in a failure to apply the network configuration. Additionally, maintaining the validated YAML syntax is useful when applying changes using Kubernetes NMState after deployment or when expanding the cluster.

Reviewer 2: This is kind of odd to state, since this is true of any invalid YAML, not sure if it's important for some reason :shrug:

Author: I was asked to add this note and I take your point maybe thre is something specific with NMState and probably OK to remind customers about importance of valid YAML.

====

.Procedure

> . Use the `networkConfig` configuration setting by adding the NMState configuration to hosts within the `install-config.yaml` file:

Reviewer 2: This didn't make sense to me and I had to re-read it a few times to see what was meant. 

I think this could be reworded to be clearer. What about

```suggestion
. Add the NMState configuration to the `networkConfig` field to hosts within the `install-config.yaml` file:
```

> . Use the `networkConfig` configuration setting by adding the NMState configuration to hosts within the `install-config.yaml` file:

Reviewer 2: Also, since this is a single step procedure, it should be an unnumbered bullet

+
[source,yaml]
----

>       hosts:

Reviewer 2: Is there a reason this is indented so far, or can it be moved left?

Reviewer: +1

      - hostname: worker-1
         interfaces:
          - name: eno1
            macAddress: 0c:42:a1:55:f3:06
          - name: eno2
            macAddress: 0c:42:a1:55:f3:07
         networkConfig: <1>
           interfaces: <2>
           - name: eno1     <3>
             type: ethernet <4>
             state: up
             mac-address: 0c:42:a1:55:f3:06
             ipv4:
               enabled: true
               dhcp: false  <5>
             ethernet:
               sr-iov:
                total-vfs: 2 <6>
             ipv6:
               enabled: false
           - name: sriov:eno1:0
             type: ethernet
             state: up  <7>
             ipv4:
               enabled: false <8>
             ipv6:
               enabled: false
           - name: sriov:eno1:1
             type: ethernet
             state: down
             min-tx-rate: 100 <9>

> 	           max-tx-rate: 200 <10>

Reviewer 3: remove spaces

             type: ethernet
             state: up
             ipv4:
               enabled: false
             ipv6:
               enabled: false
           - name: sriov:eno2:1
             type: ethernet
             state: down
           - name: bond0
             type: bond
             state: up
             min-tx-rate: 100 <9>
             max-tx-rate: 200 <10>
             link-aggregation:
               mode: active-backup <11>
               options:
                 primary: sriov:eno1:0 <12>
               port:
                - sriov:eno1:0
                - sriov:eno2:0
               ipv4:
                 address:
                 - ip: 10.19.16.57 <13>
                   prefix-length: 23
                 dhcp: false
                 enabled: true
              ipv6:
                enabled: false
              dns-resolver:
                config:
                  server:
                  - 10.11.5.160
                  - 10.2.70.215
              routes:
                config:
                - destination: 0.0.0.0/0
                  next-hop-address: 10.19.17.254
                  next-hop-interface: bond0 <14>
                  table-id: 254
----

> <1> The networkConfig field holds the network configuration of the host. It has several subfields, including `interfaces`, `dns-resolver`, and `routes`.

Reviewer 2: Same comments as the other file that had these same callouts
